File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/base/BaseBackActivity.java
package com.blogspot.e_kanivets.moneytracker.activity.base;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;

import android.view.MenuItem;

import com.blogspot.e_kanivets.moneytracker.R;

public abstract class BaseBackActivity extends BaseActivity {

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    protected Toolbar initToolbar() {
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) actionBar.setDisplayHomeAsUpEnabled(true);

        return toolbar;
    }
}

File: app/src/main/java/com/blogspot/e_kanivets/moneytracker/activity/base/BaseBackActivity.kt
package com.blogspot.e_kanivets.moneytracker.activity.base

import android.view.MenuItem
import androidx.appcompat.widget.Toolbar
import com.blogspot.e_kanivets.moneytracker.R

abstract class BaseBackActivity : BaseActivity() {

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    override fun initToolbar(): Toolbar? {
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        val actionBar = supportActionBar
        actionBar?.setDisplayHomeAsUpEnabled(true)

        return toolbar
    }
}
